### DISCUSSION TITLE
Why is my function getting called as soon as i run the program?

### DISCUSSION THREAD

**Discussion by DarkKnight263** [TomSchimansky/CustomTkinter#866] [Q&A] [answered]
hi, im trying to learn how to use python and customtkinter by editing and experimenting with the example given in the rep. I fail to understand , y is the function getting called upon running the program

PLS IGNORE THE FORMATTING ISSUE WITH THE QUESTION

`
import tkinter
import tkinter.messagebox
import customtkinter
from PIL import Image

customtkinter.set_appearance_mode("System")  # Modes: "System" (standard), "Dark", "Light"
customtkinter.set_default_color_theme("blue")  # Themes: "blue" (standard), "green", "dark-blue"
cart=[]
class App(customtkinter.CTk):
    
    def __init__(self):
        super().__init__()

        

        # configure window
        self.title("PIZZA PLACE")
        self.geometry(f"{1100}x{580}")

        # configure grid layout (4x4)
        self.grid_columnconfigure(1, weight=1)
        self.grid_columnconfigure((2, 3), weight=0)
        self.grid_rowconfigure((0, 1, 2), weight=1)

        # create sidebar frame with widgets
        self.sidebar_frame = customtkinter.CTkFrame(self, width=140, corner_radius=0)
        self.sidebar_frame.grid(row=0, column=0, rowspan=4, sticky="nsew")
        self.sidebar_frame.grid_rowconfigure(4, weight=1)
        self.sidebar_button_1 = customtkinter.CTkButton(self.sidebar_frame, command=self.sidebar_button_event("test"))
        self.sidebar_button_1.grid(row=1, column=0, padx=20, pady=10)

    def sidebar_button_event(self,a):
        print(a)

if __name__ == "__main__":
    app = App()
    app.mainloop()`

**Comment by Peanutbrain-pro** [comment, marked as answer]
use 
`command=lambda:self.sidebar_button_event("test")`

**Reply by DarkKnight263** [reply]
this works,thanks a lot man
